item-requests.md

```diff
@@ -6,36 +6,111 @@ This supports the Request a Copy feature.
+  "_links":{
+    "bitstream":{
+      "href":"http://localhost/api/tools/itemrequests/1/bitstream"
+    },
+    "item":{
+      "href":"http://localhost/api/tools/itemrequests/1/item"
+    },
+    "self":{
+      "href":"http://localhost/api/tools/itemrequests/1"
+    }
+  }
 }
 ```
 
+Item properties:
+
+  * type: always "itemrequest".  READ-ONLY
+  * id: internal unique identifier of the request.  READ-ONLY
+  * itemId: UUID of the requested Item.
+  * bitstreamId: UUID of the requested bitstream.
+  * allfiles: true if the request is for all bitstreams of the item.
+  * requestEmail: email address of the person requesting the files.
+  * requestName: Human-readable name of the person requesting the files.
+  * requestMessage: arbitrary message provided by the person requesting the files.
+  * requestDate: date that the request was recorded.  READ-ONLY.
+  * expires: date on which the request is considered expired.
+  * acceptRequest: true if the request has been granted.
+  * decisionDate: date that the request was granted or denied.  READ-ONLY.
+  * token: opaque string which uniquely identifies this request.  READ-ONLY
+
 Exposed links:
 
   * item: the item requested
   * bitstream: the bitstream requested
 
+Return codes:
+* 200 OK - if the operation succeeded
+* 404 NOT FOUND - if the token is unknown (no such request exists)
+
+## Creating a Request
+**POST /api/tools/itemrequests**
+
+Anyone may create an item request.  The Content-Type is JSON.  Example:
+```json
+{
+    "itemId": "71d9fb0c-cc36-41c1-b1d3-63887b414fca",
+    "allfiles": false,
+    "bitstreamId": "dca2dadb-7028-40a9-ab39-1a47726df7ef",
+    "requestEmail": "jqpublic@example.com",
+    "requestName": "John Q. Public",
+    "requestMessage": "Please send this to me."
+}
+```
+`bitstreamId` is ignored and may be omitted if `allfiles` is `false`.  `requestMessage` is optional.  `requestEmail` and `requestName` are ignored and may be omitted if the session is authenticated -- these fields will be filled from the session user's EPerson.  If the session is anonymous then `requestEmail` is required.  `bitstreamId` is required if `allfiles` is false.  `itemId` is always required.
+
+The response contains the complete request in JSON format, as shown in Single Request.
+
+An appropriate person will be notified that the request has been filed.
+
+Return codes:
+* 201 CREATED - if the operation succeeded
+* 401 UNAUTHORIZED - if anonymous requests are disabled and the session is unauthenticated
+* 422 UNPROCESSABLE ENTITY - if the POSTed document could not be interpreted, the Item or Bitstream could not be found, or required fields are missing
+
+## Accepting / Denying a Request
+**PUT /api/tools/itemrequests/<:token>**
+
+Anyone may accept or deny a request.  Access is controlled by keeping the token confidential.  The Content-Type is JSON.  Example:
+```json
+{
+    "acceptRequest": true,
+    "responseMessage": "Approved.  Documents attached."
+}
```

Comment:
The DSpace 6 UI has 2 fields for the response: subject and message. Perhaps that subject field should still be added?
 
![response](https://wiki.lyrasis.org/download/attachments/45548167/XMLUI-Document-copy-request-email.png?version=1&modificationDate=1386284797545&api=v2)

Reply:
Well caught.  I have added a description of "subject".